src/battle-text-parser.ts

```diff
@@ -257,28 +257,39 @@ class BattleTextParser {
 		side = side.slice(0, 2);
 		if (side === 'p1') return this.p1;
 		if (side === 'p2') return this.p2;
+		if (side === 'p3') return this.p3;
+		if (side === 'p4') return this.p4;
 		return `???side:${side}???`;
 	}
 
-	team(side: string, isFar: 0 | 1 = 0) {
+	static allyID(sideid: SideID): SideID | '' {
+		if (sideid === 'p1') return 'p3';
+		if (sideid === 'p2') return 'p4';
+		if (sideid === 'p3') return 'p1';
+		if (sideid === 'p4') return 'p2';
+		return '';
+	}
+
+	team(side: string, isFar: boolean = false) {
 		side = side.slice(0, 2);
-		if (side === (this.perspective === isFar ? 'p1' : 'p2')) {
-			return BattleText.default.team;
+		let team = [BattleText.default.team, BattleText.default.opposingTeam];
+		if (side === this.perspective || side === BattleTextParser.allyID(side as SideID)) {
+			return team[+isFar];
 		}
-		return BattleText.default.opposingTeam;
+		return team[+isFar ^ 1];
```

Comment:
```suggestion
		return team[isFar ? 0 : 1];
```

Bitwise tricks are really unreadable...